# Apps for Googleplay and App store market
We are going to build a free app on Google play and App store. Our aim of this project is to find out the app profile which attracts as many users as possible because our main source of revenue consists of in-app ads.


In [1]:
opened_file_1 = open('AppleStore.csv', encoding = 'utf8')
opened_file_2 = open('googleplaystore.csv', encoding = 'utf8')
from csv import reader
read_file_1 = reader(opened_file_1)
read_file_2 = reader(opened_file_2)
dataset_1 = list(read_file_1)
ios_heading = dataset_1[0]
ios = dataset_1[1:]
dataset_2 = list(read_file_2)
android_heading = dataset_2[0]
android = dataset_2[1:]


In [2]:
def explore_data(dataset, start, end, row_col = True):
    dataset_slice = dataset[start:end]
    for row in dataset_slice:
        print(row)
        print('\n')
    if row_col:#whether dataset includes heading row or not
        print('number_of_row:', len(dataset))
        print('number_of_column:', len(dataset[0]))
explore_data(android, 0, 3, False)


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']




In [3]:
print(ios_heading)
print('\n')
print(android_heading)

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


In [4]:
print(android[10472]) # row 10472 have an empty column

['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']


In [5]:
del android[10472] # remove row which have error

In [6]:
unique_apps = [] # count rows which is duplicated
duplicate_apps = []
for app in android:
    name = app[0]
    if name in unique_apps:
        duplicate_apps.append(name)
    else:
        unique_apps.append(name)
print(len(duplicate_apps))

1181


In [7]:
reviews_max = {} #remove duplicated data: data which have the highest number of reviews is the latest => remove the remain
for row in android:
    name = row[0]
    n_reviews = float(row[3])
    if name in reviews_max and n_reviews > reviews_max[name]:
        reviews_max[name] = n_reviews
    if name not in reviews_max:
        reviews_max[name] = n_reviews 
print(len(reviews_max))

9659


In [8]:
android_clear = []
already_added = []
for row in android:
    name = row[0]
    n_reviews = float(row[3])
    if name not in already_added and n_reviews == reviews_max[name]:
        android_clear.append(row)
        already_added.append(name)
print(len(android_clear))
        

9659


In [9]:
def English_name(name): # a function to remove non-English character ( English character have ord() < or = 127)
    for character in name:
        if ord(character) > 127:
            return False
    return True
        
English_name('Instagram')        
    

True

In [10]:
def English_name(name): # the apps name can have icons or special characters which are not considered English characters so that app names have more than 3 characters should be moved  
    n_nonEnglish = 0
    for character in name:
        if ord(character) > 127:
            n_nonEnglish +=1
    if n_nonEnglish > 3:
        return False
    else:
        return True

In [11]:
English_android = []
English_ios = []
for row in android_clear:
    name = row[0]
    if English_name(name) == True:
        English_android.append(row)

for row in ios:
    name = row[1]
    if English_name(name) == True:
        English_ios.append(row)        
print(len(English_android)) 
print(len(English_ios))

9614
6183


In [12]:
android_final = [] # remove non-free apps
ios_final = []
for row in English_android:
    price = row[7]
    if price == '0':
        android_final.append(row)
for row in English_ios:
    price = row[4]
    if price == '0.0':
        ios_final.append(row)
print(len(android_final))
print(len(ios_final))
        

8864
3222


We start to analyse the clear data to find out the suitable app profile. We will use frequency of app genres, number of reviews, number of install to decide type of app we should build in Google play and App store.

In [13]:

def fred_table(dataset, index):#create frequency table to compare the criteria of app genres
    table = {}
    total = 0
    for row in dataset:
        total += 1
        key = row[index]
        if key in table:
            table[key] += 1
            
        else:
            table[key] = 1
    percentage_table = {}
    for key in table:
        percentage = table[key] / total * 100
        percentage_table[key] = percentage
    return percentage_table
fred_table(ios_final, -5)
fred_table(android_final, 1)
fred_table(android_final, -4)    

def display_table(a_dict): # Build a function sort data in descending order
    sorted_table = []
    for key in a_dict:
        freq = a_dict[key] 
        a_tuple = (freq, key)
        sorted_table.append(a_tuple)
    return(sorted(sorted_table, reverse = True))
print(display_table(fred_table(ios_final, -5)))
print('\n')
print(display_table(fred_table(android_final, 1))) 
print('\n')
print(display_table(fred_table(android_final, -4)))        
            
        

[(58.16263190564867, 'Games'), (7.883302296710118, 'Entertainment'), (4.9658597144630665, 'Photo & Video'), (3.662321539416512, 'Education'), (3.2898820608317814, 'Social Networking'), (2.60707635009311, 'Shopping'), (2.5139664804469275, 'Utilities'), (2.1415270018621975, 'Sports'), (2.0484171322160147, 'Music'), (2.0173805090006205, 'Health & Fitness'), (1.7380509000620732, 'Productivity'), (1.5828677839851024, 'Lifestyle'), (1.3345747982619491, 'News'), (1.2414649286157666, 'Travel'), (1.1173184357541899, 'Finance'), (0.8690254500310366, 'Weather'), (0.8069522036002483, 'Food & Drink'), (0.5586592178770949, 'Reference'), (0.5276225946617008, 'Business'), (0.4345127250155183, 'Book'), (0.186219739292365, 'Navigation'), (0.186219739292365, 'Medical'), (0.12414649286157665, 'Catalogs')]


[(18.907942238267147, 'FAMILY'), (9.724729241877256, 'GAME'), (8.461191335740072, 'TOOLS'), (4.591606498194946, 'BUSINESS'), (3.9034296028880866, 'LIFESTYLE'), (3.892148014440433, 'PRODUCTIVITY'), (3.7

In Appstore, games apps are the most common, make up more than 58%. The percentage of entertainment apps is 7.88%, follow by the
Photo&Video apps and education apps, nearly 5% and 3.66% respectively.
In Google play market, with regard to category, family are the most common apps, 18.9%, follow by game app, at 9.72%. The third position is tools apps account for 8.46%. With regard to genre, tools apps is the most common making up  8.44%, follow by entertainment and education apps
In both Appstore and Google play market, apps for entertainment make up the majority.





In [14]:
print(fred_table(ios_final, -5))

{'Social Networking': 3.2898820608317814, 'Photo & Video': 4.9658597144630665, 'Games': 58.16263190564867, 'Music': 2.0484171322160147, 'Reference': 0.5586592178770949, 'Health & Fitness': 2.0173805090006205, 'Weather': 0.8690254500310366, 'Utilities': 2.5139664804469275, 'Travel': 1.2414649286157666, 'Shopping': 2.60707635009311, 'News': 1.3345747982619491, 'Navigation': 0.186219739292365, 'Lifestyle': 1.5828677839851024, 'Entertainment': 7.883302296710118, 'Food & Drink': 0.8069522036002483, 'Sports': 2.1415270018621975, 'Book': 0.4345127250155183, 'Finance': 1.1173184357541899, 'Education': 3.662321539416512, 'Productivity': 1.7380509000620732, 'Business': 0.5276225946617008, 'Catalogs': 0.12414649286157665, 'Medical': 0.186219739292365}


In [20]:
#We want to find out which genre is popular in Appstore but we do not have number of install
#count the average of number of users rating for apps in Appstore
ios_genre = fred_table(ios_final, -5)
sort_ave_genre = []
for genre in ios_genre:
    total = 0
    len_genre = 0
    for row in ios_final:
        prime_genre = row[-5]
        if prime_genre == genre:
            total += float(row[5])
            len_genre += 1
    ave_genre = total / len_genre
    tuple = (ave_genre, genre)
    sort_ave_genre.append(tuple)
print(sorted(sort_ave_genre, reverse = True))

    
         
    
    
    
    

[(86090.33333333333, 'Navigation'), (74942.11111111111, 'Reference'), (71548.34905660378, 'Social Networking'), (57326.530303030304, 'Music'), (52279.892857142855, 'Weather'), (39758.5, 'Book'), (33333.92307692308, 'Food & Drink'), (31467.944444444445, 'Finance'), (28441.54375, 'Photo & Video'), (28243.8, 'Travel'), (26919.690476190477, 'Shopping'), (23298.015384615384, 'Health & Fitness'), (23008.898550724636, 'Sports'), (22788.6696905016, 'Games'), (21248.023255813954, 'News'), (21028.410714285714, 'Productivity'), (18684.456790123455, 'Utilities'), (16485.764705882353, 'Lifestyle'), (14029.830708661417, 'Entertainment'), (7491.117647058823, 'Business'), (7003.983050847458, 'Education'), (4004.0, 'Catalogs'), (612.0, 'Medical')]


Navigation is the most popular app genre in Appstore

In [16]:
print(fred_table(android_final, 5))
print('\n')
print(fred_table(android_final, 1))

{'10,000+': 10.198555956678701, '5,000,000+': 6.825361010830325, '50,000,000+': 2.3014440433213, '100,000+': 11.552346570397113, '50,000+': 4.7721119133574, '1,000,000+': 15.726534296028879, '10,000,000+': 10.548285198555957, '5,000+': 4.512635379061372, '500,000+': 5.561823104693141, '1,000,000,000+': 0.22563176895306858, '100,000,000+': 2.1322202166064983, '1,000+': 8.393501805054152, '500,000,000+': 0.2707581227436823, '500+': 3.2490974729241873, '100+': 6.915613718411552, '50+': 1.917870036101083, '10+': 3.5424187725631766, '1+': 0.5076714801444043, '5+': 0.78971119133574, '0+': 0.04512635379061372, '0': 0.01128158844765343}


{'ART_AND_DESIGN': 0.6430505415162455, 'AUTO_AND_VEHICLES': 0.9250902527075812, 'BEAUTY': 0.5979241877256317, 'BOOKS_AND_REFERENCE': 2.1435018050541514, 'BUSINESS': 4.591606498194946, 'COMICS': 0.6204873646209386, 'COMMUNICATION': 3.2378158844765346, 'DATING': 1.861462093862816, 'EDUCATION': 1.1620036101083033, 'ENTERTAINMENT': 0.9589350180505415, 'EVENTS': 0

In [17]:
android_fl = []
for row in android_final:
    install = row[5]
    install = install.replace("+" ,"")
    install = install.replace(',', '')
    row[5] = install # We consider 10000+ people install apps as 100000 people and remove '+' and ',' to convert to float. 
    android_fl.append(row)# updated dataset
    
print(android_fl[1])

    

['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5000000', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


In [18]:
fred_table(android_fl, 1)

{'ART_AND_DESIGN': 0.6430505415162455,
 'AUTO_AND_VEHICLES': 0.9250902527075812,
 'BEAUTY': 0.5979241877256317,
 'BOOKS_AND_REFERENCE': 2.1435018050541514,
 'BUSINESS': 4.591606498194946,
 'COMICS': 0.6204873646209386,
 'COMMUNICATION': 3.2378158844765346,
 'DATING': 1.861462093862816,
 'EDUCATION': 1.1620036101083033,
 'ENTERTAINMENT': 0.9589350180505415,
 'EVENTS': 0.7107400722021661,
 'FINANCE': 3.7003610108303246,
 'FOOD_AND_DRINK': 1.2409747292418771,
 'HEALTH_AND_FITNESS': 3.0798736462093865,
 'HOUSE_AND_HOME': 0.8235559566787004,
 'LIBRARIES_AND_DEMO': 0.9363718411552346,
 'LIFESTYLE': 3.9034296028880866,
 'GAME': 9.724729241877256,
 'FAMILY': 18.907942238267147,
 'MEDICAL': 3.531137184115524,
 'SOCIAL': 2.6624548736462095,
 'SHOPPING': 2.2450361010830324,
 'PHOTOGRAPHY': 2.944494584837545,
 'SPORTS': 3.395758122743682,
 'TRAVEL_AND_LOCAL': 2.33528880866426,
 'TOOLS': 8.461191335740072,
 'PERSONALIZATION': 3.3167870036101084,
 'PRODUCTIVITY': 3.892148014440433,
 'PARENTING': 0.6

In [21]:
app_install = []
for category in fred_table(android_fl, 1):
    len_apps = 0
    total_install = 0
    for row in android_fl:
        n_install = float(row[5])
        if row[1] == category:
            len_apps += 1
            total_install += n_install
    avg_install = total_install / len_apps
    a_tuple = (avg_install, category)
    app_install.append(a_tuple)
print(sorted(app_install, reverse = True))
  


[(38456119.167247385, 'COMMUNICATION'), (24727872.452830188, 'VIDEO_PLAYERS'), (23253652.127118643, 'SOCIAL'), (17840110.40229885, 'PHOTOGRAPHY'), (16787331.344927534, 'PRODUCTIVITY'), (15588015.603248259, 'GAME'), (13984077.710144928, 'TRAVEL_AND_LOCAL'), (11640705.88235294, 'ENTERTAINMENT'), (10801391.298666667, 'TOOLS'), (9549178.467741935, 'NEWS_AND_MAGAZINES'), (8767811.894736841, 'BOOKS_AND_REFERENCE'), (7036877.311557789, 'SHOPPING'), (5201482.6122448975, 'PERSONALIZATION'), (5074486.197183099, 'WEATHER'), (4188821.9853479853, 'HEALTH_AND_FITNESS'), (4056941.7741935486, 'MAPS_AND_NAVIGATION'), (3695641.8198090694, 'FAMILY'), (3638640.1428571427, 'SPORTS'), (1986335.0877192982, 'ART_AND_DESIGN'), (1924897.7363636363, 'FOOD_AND_DRINK'), (1833495.145631068, 'EDUCATION'), (1712290.1474201474, 'BUSINESS'), (1437816.2687861272, 'LIFESTYLE'), (1387692.475609756, 'FINANCE'), (1331540.5616438356, 'HOUSE_AND_HOME'), (854028.8303030303, 'DATING'), (817657.2727272727, 'COMICS'), (647317.817

Communication is the app category which have the most installing in Google play market